network/oauth_client.go

```diff
@@ -39,9 +40,7 @@ func NewOAuthClient(target, username, password string, insecureSkipVerify bool)
 				Timeout:   5 * time.Second,
 				KeepAlive: 30 * time.Second,
 			}).Dial,
-			ResponseHeaderTimeout: 15 * time.Minute,
```

Comment:
We don't necessarily know how this is going to work out but we are ok with trying an overall timeout. This ResponseHeader timeout was added due to issues we were seeing on both Azure/GCP. It may be added back.